### ISSUE TITLE
Contains on Sub query with entity type generates invalid SQL

### ISSUE THREAD

**Issue by mikary** [aspnet/EntityFramework#4355]
The following code throws a SqlException: `Subquery returned more than 1 value. This is not permitted when the subquery follows =, !=, <, <= , >, >= or when the subquery is used as an expression.`

```cs
        [ConditionalFact]
        public virtual void Where_subquery_on_collection_of_entity_type()
        {
            AssertQuery<Product, OrderDetail>((pr, od) =>
                pr.Where(
                        p => od
                            .Where(o => o.ProductID == p.ProductID)
                            .Contains(od.FirstOrDefault())));
        }
```

With the generated SQL:
```sql
SELECT [p].[ProductID], [p].[Discontinued], [p].[ProductName], [p].[UnitsInStock]
FROM [Products] AS [p]
WHERE (
    SELECT 1
    FROM [Order Details] AS [o]
    WHERE [o].[ProductID] = [p].[ProductID]
) = 1
```

**Comment by divega** [comment]
@mikary wouldn't this query just return incorrect results if there is only zero or one row in 
`[Order Details]` matching each row in `Products`? I.e. I believe the SQL isn't really invalid but may fail depending on how the data in the database looks like, correct?

**Comment by mikary** [comment]
Just tested a few variations and it appears you are correct.

Single result (returns all products)
``sql
SELECT [p].[ProductID], [p].[Discontinued], [p].[ProductName], [p].[UnitsInStock]
FROM [Products] AS [p]
WHERE (
    SELECT 1
    FROM [Order Details] AS [o]
    WHERE [o].[OrderID] = 10285 AND [o].[ProductID] = 1
) = 1
```

No result (returns no products, technically the correct result)
```sql
SELECT [p].[ProductID], [p].[Discontinued], [p].[ProductName], [p].[UnitsInStock]
FROM [Products] AS [p]
WHERE (
    SELECT 1
    FROM [Order Details] AS [o]
    WHERE [o].[OrderID] = 0 AND [o].[ProductID] = 1
) = 1
```

**Comment by mikary** [comment]
This issue is very similar to #153, but doesn't involve navigations / has a different shape to the expression

**Comment by maumar** [comment]
Poaching this per conversation with @anpet

**Comment by maumar** [comment]
fixed in 63272b0b7b5fc888b06c6993c1815e8ca780fada
